### ISSUE TITLE
NullPointerException caused by CompilerFlagsProvider

### ISSUE THREAD

**Issue by carymrobbins** [carymrobbins/intellij-haskforce#266]
It seems `provider.getFlags` is returning null, causing an NPE.  This should be fixed.  Note that this occurred when working on HaskForce with no Haskell Tools configured.

```
null
java.lang.NullPointerException
	at java.util.Collections.addAll(Collections.java:5400)
	at com.haskforce.codeInsight.HaskellCompletionCacheLoader.com$haskforce$codeInsight$HaskellCompletionCacheLoader$$putStrings(HaskellCompletionCacheLoader.scala:86)
	at com.haskforce.codeInsight.HaskellCompletionCacheLoader$$anonfun$updateCache$1$$anonfun$apply$mcV$sp$1.apply(HaskellCompletionCacheLoader.scala:62)
	at com.haskforce.codeInsight.HaskellCompletionCacheLoader$$anonfun$updateCache$1$$anonfun$apply$mcV$sp$1.apply(HaskellCompletionCacheLoader.scala:61)
	at scala.Option.foreach(Option.scala:257)
	at com.haskforce.codeInsight.HaskellCompletionCacheLoader$$anonfun$updateCache$1.apply$mcV$sp(HaskellCompletionCacheLoader.scala:61)
	at com.haskforce.utils.SAMUtils$$anon$3.run(SAMUtils.scala:18)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:369)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by carymrobbins** [comment]
Possibly related -

```
null
com.haskforce.highlighting.annotation.external.GhcModi$CheckParseError
	at com.haskforce.highlighting.annotation.external.GhcModi.handleCheck(GhcModi.java:165)
	at com.haskforce.highlighting.annotation.external.GhcModi.unsafeCheck(GhcModi.java:122)
	at com.haskforce.highlighting.annotation.external.impl.GhcModiCompileProblemsProvider.getProblems(GhcModiCompileProblemsProvider.scala:18)
	at com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator$State$$anonfun$buildForFile$1$$anonfun$apply$1.apply(HaskellExternalAnnotator.scala:94)
	at com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator$State$$anonfun$buildForFile$1$$anonfun$apply$1.apply(HaskellExternalAnnotator.scala:94)
	at scala.Option.map(Option.scala:146)
	at com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator$State$$anonfun$buildForFile$1.apply(HaskellExternalAnnotator.scala:94)
	at com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator$State$$anonfun$buildForFile$1.apply(HaskellExternalAnnotator.scala:94)
	at scala.collection.immutable.List.flatMap(List.scala:327)
	at com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator$State$.buildForFile(HaskellExternalAnnotator.scala:94)
	at com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator.doAnnotate(HaskellExternalAnnotator.scala:44)
	at com.haskforce.highlighting.annotation.external.HaskellExternalAnnotator.doAnnotate(HaskellExternalAnnotator.scala:19)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.d(ExternalToolPass.java:218)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.access$500(ExternalToolPass.java:44)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:160)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:333)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:323)
	at com.intellij.util.ui.update.MergingUpdateQueue$3.run(MergingUpdateQueue.java:267)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:282)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:234)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:238)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:378)
	at com.intellij.util.Alarm$Request.run(Alarm.java:389)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:227)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:187)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by askg7dawn** [comment]
Hi. i'm trying to use your HaskForce, but "IDE Fatal Errors (Exception in plugin `SBT Executor`)" was occurred.
It seems to be related to 'NullPointerException', so i want to share the error message which i received.

```
null
com.haskforce.highlighting.annotation.external.GhcModi$ExecError
	at com.haskforce.highlighting.annotation.external.GhcModi.write(GhcModi.java:325)
	at com.haskforce.highlighting.annotation.external.GhcModi.interact(GhcModi.java:292)
	at com.haskforce.highlighting.annotation.external.GhcModi.exec(GhcModi.java:242)
	at com.haskforce.highlighting.annotation.external.GhcModi.simpleExec(GhcModi.java:208)
	at com.haskforce.highlighting.annotation.external.GhcModi.simpleExecToLines(GhcModi.java:220)
	at com.haskforce.highlighting.annotation.external.GhcModi.simpleExecToLinesOrEmpty(GhcModi.java:226)
	at com.haskforce.highlighting.annotation.external.GhcModi.unsafeFlag(GhcModi.java:132)
	at com.haskforce.codeInsight.GhcModiCompilerFlagsProvider.getFlags(CompilerFlagsProviderFactory.scala:25)
	at com.haskforce.codeInsight.HaskellCompletionCacheLoader$$anonfun$updateCache$1$$anonfun$apply$mcV$sp$1.apply(HaskellCompletionCacheLoader.scala:62)
	at com.haskforce.codeInsight.HaskellCompletionCacheLoader$$anonfun$updateCache$1$$anonfun$apply$mcV$sp$1.apply(HaskellCompletionCacheLoader.scala:61)
	at scala.Option.foreach(Option.scala:257)
	at com.haskforce.codeInsight.HaskellCompletionCacheLoader$$anonfun$updateCache$1.apply$mcV$sp(HaskellCompletionCacheLoader.scala:61)
	at com.haskforce.utils.SAMUtils$$anon$3.run(SAMUtils.scala:18)
	at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:369)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by carymrobbins** [comment]
This will be fixed in the next release **v0.3-beta.32**.
